src/figuresClasses.ts

```diff
@@ -1,19 +1,86 @@
-export interface Figure {
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
 }
 
-export class Triangle {
+export interface Figure {
+  shape: Shape
+  color: Color
 
+  getArea(): number
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  public shape: Shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const maxSideLength = Math.max(a, b, c);
+
+    if ((a || b || c) <= 0) {
+      throw new Error('your error message');
+    }
+
+    if (maxSideLength >= a + b + c - maxSideLength) {
+      throw new Error('your error message');
+    }
+  }
 
+  public getArea(): number {
+    const { a, b, c } = this;
+    const p: number = (a + b + c) / 2;
+    const sqrt: number = Math.sqrt(p * (p - a) * (p - b) * (p - c));
+
+    return Math.floor((sqrt * 100)) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  public shape: Shape = Shape.Circle;
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('your error message');
+    }
+  }
 
+  public getArea(): number {
+    return Math.floor(Math.PI * (this.radius ** 2) * 100) / 100;
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  public shape: Shape = Shape.Rectangle;
+
+  constructor(
+    public color: Color,
+    public width: number,
+    public height: number,
+  ) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('your error message');
+    }
+  }
+
+  public getArea(): number {
+    return this.height * this.width;
```

Comment:
Move this to separate variable

Reply:
There is no point in putting it in a separate variable.